Appsilon/rhino-showcase
# fix: add updated .shp with Crimea in Ukraine

### Changes

As raised by @vanhry, the borders for Crimea were being reflected under Russia. The shapefile is now updated with [this file](https://hub.arcgis.com/datasets/esri::world-countries-generalized/explore?location=-0.197969%2C0.000000%2C2.00) from ESRI and ArcGIS.

<img width="1007" alt="Screenshot 2023-02-14 at 12 20 26 PM" src="https://user-images.githubusercontent.com/26517718/218662855-61d49561-a910-468b-aad5-be9ead0761a9.png">

To make it suitable to be used with the app, I had to do some wrangling to get the `cnt_LAT`, `cnt_LON` and `ISO3c` columns into this file and resave it using `st_write()`. This is elaborated further in the Future Reference heading below.

### To Test

- Run the app as you would
- Verify borders for Crimea
- If possible, do verify other borders that may be erroneous or changed

### Future Reference

For reference, the `sf`/`dplyr` pipeline used for this was:

```
map_data_ <- map_data_ |>
    dplyr::mutate(ISO3c = countrycode::countrycode(ISO, origin = "iso2c", destination = "iso3c"))

sf::st_join(map_data_, map_data |>
    dplyr::select(ISO3c, cnt_LON, cnt_LAT)) |>
    dplyr::filter(ISO3c.x == ISO3c.y)

sf::st_join(map_data_, map_data |>
    dplyr::select(ISO3c, cnt_LON, cnt_LAT)) |>
    dplyr::filter(ISO3c.x == ISO3c.y) |>
    dplyr::select(-ISO3c.y, AFF_ISO, COUNTRYAFF) |>
    dplyr::rename(ISO3c = ISO3c.x) |>
    sf::st_write("app/data/new_map_data.shp")
```

Where `map_data_` is the shapefile downloaded from the link above, read using `sf::st_read()` and the `map_data` object is read from the older data.
